clients/filesystem-hadoop/src/test/java/com/datastrato/gravitino/filesystem/hadoop/TestGravitinoVirtualFileSystem.java

```diff
@@ -0,0 +1,377 @@
+
+    // test gvfs, should not get the same fs
+    String anotherFileset = "fileset_2";
+    Path localPath = FileSystemTestUtils.createLocalPrefix(catalogName, schemaName, anotherFileset);
+
+    mockFilesetDTO(metalakeName, catalogName, schemaName, anotherFileset, localPath.toString());
+
+    Path diffGvfsPath =
+        FileSystemTestUtils.createGvfsPrefix(catalogName, schemaName, anotherFileset);
+    FileSystem anotherGvfs = diffGvfsPath.getFileSystem(conf);
+    assertNotEquals(anotherGvfs, gravitinoFileSystem);
+
+    // test proxyed actual local, should not get the same fs
+    FileSystem proxyLocal =
+        Objects.requireNonNull(
+                ((GravitinoVirtualFileSystem) gravitinoFileSystem)
+                    .getFilesetCache()
+                    .getIfPresent(
+                        NameIdentifier.of(metalakeName, catalogName, schemaName, filesetName)))
+            .getFileSystem();
+
+    Path diffLocalPath =
+        FileSystemTestUtils.createLocalPrefix(catalogName, schemaName, anotherFileset);
+    FileSystem local = diffLocalPath.getFileSystem(conf);
+
+    assertNotEquals(local, proxyLocal);
+  }
+
+  @Test
+  public void testCreate() throws IOException {
+    FileSystemTestUtils.create(gvfsPath, gravitinoFileSystem);
+    assertTrue(gravitinoFileSystem.exists(gvfsPath));
+    gravitinoFileSystem.delete(gvfsPath, true);
+
+    // mock the unverified fileset with diff name
+    String unverifiedCreateFilesetName = "fileset_create_unverified";
+    Path unverifiedCreateFilesetPath =
+        FileSystemTestUtils.createGvfsPrefix(catalogName, schemaName, unverifiedCreateFilesetName);
+    assertThrows(
+        RuntimeException.class,
+        () -> FileSystemTestUtils.create(unverifiedCreateFilesetPath, gravitinoFileSystem));
+
+    // mock the not correct prefix path
+    Path localPrefixPath = FileSystemTestUtils.createLocalPrefix(catalogName, schemaName, "test");
+    assertThrows(
+        RuntimeException.class,
+        () -> FileSystemTestUtils.create(localPrefixPath, gravitinoFileSystem));
+
+    // test fileset mount the single file
+    Path localSingleFilePath =
+        FileSystemTestUtils.createLocalFilePath(catalogName + "/files/test.txt");
+    Path filesetMountPath =
+        FileSystemTestUtils.createGvfsPrefix(catalogName, schemaName, "fileset_single_file");
+    mockFilesetDTO(
+        metalakeName,
+        catalogName,
+        schemaName,
+        "fileset_single_file",
+        localSingleFilePath.toString());
+    FileSystemTestUtils.create(filesetMountPath, gravitinoFileSystem);
+    assertTrue(gravitinoFileSystem.exists(filesetMountPath));
+    assertTrue(gravitinoFileSystem.isFile(filesetMountPath));
+    gravitinoFileSystem.delete(filesetMountPath, true);
+    assertFalse(localFileSystem.exists(localSingleFilePath));
+  }
+
+  @Test
+  @Disabled("Append operation is not supported in LocalFileSystem. We can't test it now.")
```

Comment:
The `Append` operation is not supported in `LocalFileSystem`, maybe we should complete this test in the integration test.